cognite/neat/graph/extractors/_dms.py

```diff
@@ -0,0 +1,162 @@
+        """Create an extractor from a set of views.
+
+        Args:
+            client: The Cognite client to use.
+            views: The views to extract.
+            limit: The maximum number of instances to extract.
+        """
+        return cls(cls._InstanceIterator(client, views), total=None, limit=limit)
+
+    def extract(self) -> Iterable[Triple]:
+        for count, item in enumerate(self.items, 1):
+            if self.limit and count > self.limit:
+                break
+            yield from self._extract_instance(item)
+
+    def _extract_instance(self, instance: Instance) -> Iterable[Triple]:
+        if isinstance(instance, dm.Edge):
+            if not instance.properties:
+                yield (
+                    self._as_uri_ref(instance.start_node),
+                    self._as_uri_ref(instance.type),
+                    self._as_uri_ref(instance.end_node),
+                )
+                return
+            else:
+                # If the edge has properties, we create a node for the edge and connect it to the start and end nodes.
+                id_ = self._as_uri_ref(instance)
+                yield id_, RDF.type, self._as_uri_ref(instance.type)
+                yield id_, RDF.type, self._as_uri_ref(dm.DirectRelationReference(instance.space, "Edge"))
+                yield (
+                    id_,
+                    self._as_uri_ref(dm.DirectRelationReference(instance.space, "startNode")),
+                    self._as_uri_ref(instance.start_node),
+                )
+                yield (
+                    id_,
+                    self._as_uri_ref(dm.DirectRelationReference(instance.space, "endNode")),
+                    self._as_uri_ref(instance.end_node),
+                )
+
+        elif isinstance(instance, dm.Node):
+            id_ = self._as_uri_ref(instance)
+            if instance.type:
+                type_ = self._as_uri_ref(cast(dm.DirectRelationReference, instance.type))
+            else:
+                type_ = self._as_uri_ref(dm.DirectRelationReference(instance.space, "Node"))
+
+            yield id_, RDF.type, type_
+        else:
+            raise NotImplementedError(f"Unknown instance type {type(instance)}")
+
+        for view_id, properties in instance.properties.items():
+            namespace = self._get_namespace(view_id.space)
+            for key, value in properties.items():
+                for object_ in self._get_objects(value):
+                    yield id_, namespace[key], object_
+
+    def _get_objects(self, value: PropertyValue) -> Iterable[Literal | URIRef]:
+        if isinstance(value, str | float | bool | int):
+            yield Literal(value)
+        elif isinstance(value, dict) and "space" in value and "externalId" in value:
+            yield self._as_uri_ref(dm.DirectRelationReference.load(value))
+        elif isinstance(value, dict):
+            # This object is a json object.
+            yield Literal(str(value))
```

Comment:
@nikokaoja Need some input here

Reply:
Why we are doing this? The reason being that when we are loading this to Nodes it will be properly resolved as JSON (i.e. python dict), otherwise you lose datatype and property value gets provided as string.

Reply:
here in loader: https://github.com/cognitedata/neat/blob/ef613bbee327b26d113ba3793654225a810f4702/cognite/neat/graph/loaders/_rdf2dms.py#L198

Reply:
Also helps with inference of data model